SYCL/InvokeSimd/ESIMD_to_SPMD_error_call_vadd_1d.cpp

```diff
@@ -0,0 +1,218 @@
+  if constexpr (use_vector) {
+    return SPMD_CALLEE_doVadd_vector(va, vb);
+    // CHECK: {{.*}}error: SYCL device function cannot be called from an ESIMD context
+  }
+
+  esimd::simd<float, VL> vc;
+  for (int i = 0; i < VL; ++i) {
+    float a = va[i];
+    float b = vb[i];
+    // This should cause an error.
+    vc[i] = SPMD_CALLEE_doVadd_scalar(a, b);
+    // CHECK: {{.*}}error: SYCL device function cannot be called from an ESIMD context
+  }
+  return vc;
+}
+
+[[intel::device_indirectly_callable]] SYCL_EXTERNAL
+    simd<float, VL> __regcall SIMD_CALLEE_doVadd(
+        simd<float, VL> va, simd<float, VL> vb) SYCL_ESIMD_FUNCTION;
+
+class ESIMDSelector : public device_selector {
+  // Require GPU device unless HOST is requested in SYCL_DEVICE_FILTER env
+  virtual int operator()(const device &device) const {
+    if (const char *dev_filter = getenv("SYCL_DEVICE_FILTER")) {
+      std::string filter_string(dev_filter);
+      if (filter_string.find("gpu") != std::string::npos)
+        return device.is_gpu() ? 1000 : -1;
+      if (filter_string.find("host") != std::string::npos)
+        return device.is_host() ? 1000 : -1;
+      std::cerr
+          << "Supported 'SYCL_DEVICE_FILTER' env var values are 'gpu' and "
+             "'host', '"
+          << filter_string << "' does not contain such substrings.\n";
+      return -1;
+    }
+    // If "SYCL_DEVICE_FILTER" not defined, only allow gpu device
+    return device.is_gpu() ? 1000 : -1;
+  }
+};
+
+inline auto createExceptionHandler() {
+  return [](exception_list l) {
+    for (auto ep : l) {
+      try {
+        std::rethrow_exception(ep);
+      } catch (sycl::exception &e0) {
+        std::cout << "sycl::exception: " << e0.what() << std::endl;
+      } catch (std::exception &e) {
+        std::cout << "std::exception: " << e.what() << std::endl;
+      } catch (...) {
+        std::cout << "generic exception\n";
+      }
+    }
+  };
+}
+
+#ifndef INVOKE_SIMD
+#define INVOKE_SIMD 1
+#endif
+
+constexpr bool use_invoke_simd = INVOKE_SIMD != 0;
+
+int main(void) {
+  float *A = new float[Size];
+  float *B = new float[Size];
+  float *C = new float[Size];
+
+  for (unsigned i = 0; i < Size; ++i) {
+    A[i] = B[i] = i;
+    C[i] = 0.0f;
+  }
+
+  try {
+    buffer<float, 1> bufa(A, range<1>(Size));
+    buffer<float, 1> bufb(B, range<1>(Size));
+    buffer<float, 1> bufc(C, range<1>(Size));
+
+    // We need that many workgroups
+    sycl::range<1> GlobalRange{Size};
+
+    // We need that many threads in each group
+    sycl::range<1> LocalRange{VL};
+
+    queue q(ESIMDSelector{}, createExceptionHandler());
+
+    auto dev = q.get_device();
+    std::cout << "Running on " << dev.get_info<info::device::name>() << "\n";
+
+    auto e = q.submit([&](handler &cgh) {
+      auto PA = bufa.get_access<access::mode::read>(cgh);
+      auto PB = bufb.get_access<access::mode::read>(cgh);
+      auto PC = bufc.get_access<access::mode::write>(cgh);
+
+      cgh.parallel_for<class Test>(
```

Comment:
Please minimize the test. For negative test it is enough to use SYCL_EXTERNAL ESIMD functions as a starting point, w/o kernels, queues, exception handlers etc.
```
SYCL_EXTERNAL void foo() SYCL_ESIMD_FUNCTION {
  // test code
}
```

Reply:
Moved error message checking tests to clang_run_error directory and minimized them as much as possible.